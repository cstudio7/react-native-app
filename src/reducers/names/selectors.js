export const getNames = (state, { route, screen }) => {
  const names = state.names[route.key];
  if (screen === 'Favorites') {
    return names.filter(name => name.isFavorite);
  }
  return names;
};