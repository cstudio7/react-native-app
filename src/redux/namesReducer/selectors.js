export const filterFavorites = names => {
  const filter = genderNames => genderNames.filter(name => name.isFavorite);
  const female = filter(names.female);
  const male = filter(names.male);
  return { female, male };
};
