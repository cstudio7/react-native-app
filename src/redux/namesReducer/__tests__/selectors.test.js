import { filterFavorites } from '../selectors';

it('filters favorited names', () => {
  const names = {
    female: [],
    male: [
      { name: 'artur', isFavorite: true },
      { name: 'alex', isFavorite: false },
      { name: 'pierre' }
    ]
  };

  expect(filterFavorites(names)).toMatchSnapshot();
});
