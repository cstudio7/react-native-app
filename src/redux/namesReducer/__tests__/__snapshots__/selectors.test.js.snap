// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`filters favorited names 1`] = `
Object {
  "female": Array [],
  "male": Array [
    Object {
      "isFavorite": true,
      "name": "artur",
    },
  ],
}
`;
